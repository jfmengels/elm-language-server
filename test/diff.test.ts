import "reflect-metadata";
import { container } from "tsyringe";
import { URI } from "vscode-uri";
import { formatText } from "../src/util/diff";
import { Connection } from "vscode-languageserver";

container.register("Connection", {
  useValue: {
    console: {
      info: (a: string): void => {
        // console.log(a);
      },
      warn: (a: string): void => {
        // console.log(a);
      },
      error: (a: string): void => {
        // console.log(a);
      },
    },
    window: {
      showErrorMessage: (a: string): void => {
        console.log(a);
      },
    },
  },
});
describe("test formatting", () => {
  const connection = container.resolve<Connection>("Connection");
  const pathUri = URI.file(__dirname);

  test("normal format gives correct result", async () => {
    const result = await formatText(
      pathUri,
      "elm-format",
      "module Test",
      connection,
    );

    expect(result).toMatchSnapshot();
  });
});
