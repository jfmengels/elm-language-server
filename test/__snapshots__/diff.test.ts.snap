// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test formatting normal format gives correct result 1`] = `
Array [
  Object {
    "newText": " exposing (..)
",
    "range": Object {
      "end": Object {
        "character": 11,
        "line": 0,
      },
      "start": Object {
        "character": 11,
        "line": 0,
      },
    },
  },
]
`;
